pandas/tests/util/test_assert_produces_warning.py

```diff
@@ -94,19 +94,46 @@ def test_catch_warning_category_and_match(category, message, match):
         warnings.warn(message, category)
 
 
-@pytest.mark.parametrize(
-    "message, match",
-    [
-        ("Warning message", "Not this message"),
-        ("Warning message", "warning"),
-        ("Warning message", r"\d+"),
-    ],
-)
-def test_fail_to_match(category, message, match):
-    msg = f"Did not see warning {repr(category.__name__)} matching"
-    with pytest.raises(AssertionError, match=msg):
+def test_fail_to_match_runtime_warning():
+    category = RuntimeWarning
+    match = "Did not see this warning"
+    unmatched = (
+        r"Did not see warning 'RuntimeWarning' matching 'Did not see this warning'. "
```

Comment:
Would it be more readable if you split the message in lines this way:
```
    r"Did not see warning 'RuntimeWarning' matching 'Did not see this warning'. "
    r"The emitted warning messages are "
    r"\[RuntimeWarning\('This is not a match.'\), "
    r"RuntimeWarning\('Another unmatched warning.'\)\]"
```

Reply:
Yeah, I think that is better. I'll do that.

Reply:
Great, not it looks much cleaner here and in other places as well!